cidc_api/models/templates/sync_schemas.py

```diff
@@ -0,0 +1,226 @@
+from sqlalchemy.orm.session import Session
+from typing import Dict, List, Optional, Tuple
+
+from ..models import TrialMetadata, UploadJobs
+from .trial_metadata import (
+    ClinicalTrial,
+    Cohort,
+    CollectionEvent,
+    Participant,
+    Sample,
+    Shipment,
+)
+from .utils import with_default_session
+
+
+def _make_sample_to_shipment_map(trial_id: str, session: Session) -> Dict[str, str]:
+    """
+    For a given trial, generate a mapping from a cimac_id to the manifest_id it was loaded.
+    Uses the UploadJobs where each patch contains only a single shipment and its samples.
+    """
+    sample_to_shipment_map = {}
+
+    uploads = (
+        session.query(UploadJobs)
+        .filter(UploadJobs.trial_id == trial_id, UploadJobs.multifile == False)
+        .all()
+    )
+    for upload in uploads:
```

Comment:
@scvannost How does this deal with multiple uploads of the same shipping manifest which we see frequently in our existing database? Just curious.

Reply:
It will use the "last" name, which one will depend on the return order of the query
I could specify an `order by _created` to always use the most recent one